extensions/skins/conversation_v1/Conversation.js

```diff
@@ -589,7 +588,7 @@ oppia.directive('conversationSkin', [function() {
       });
 
       $window.addEventListener('beforeunload', function(e) {
-        if (hasInteractedAtLeastOnce && !$scope.explorationCompleted &&
+        if (hasInteractedAtLeastOnce && !$scope.isOnFinalCard() &&
```

Comment:
Or, actually, it may be simpler to just check directly here whether the last card in the transcript  (as opposed to the currently-active card) has a terminal interaction. Then you don't need a new variable.